### ISSUE TITLE
FocusTrap not working on Dialog when last focusable component is RadzenDatePicker

### ISSUE THREAD

**Issue by Molinware** [radzenhq/radzen-blazor#1024]
**Describe the bug**
This issue implements focus trap on dialog: https://github.com/radzenhq/radzen-blazor/issues/923.
The focus trap does not work if you open a dialog with the last focusable element being a RadzenDateTimePicker.

**To Reproduce**
```
var result = await DialogService.OpenAsync("Simple Dialog", ds =>
    @<div>
    <RadzenButton Text="Ok" Click="() => ds.Close(true)" Style="width: 80px;" />
    <RadzenButton Text="Cancel" Click="() => ds.Close(false)" ButtonStyle="ButtonStyle.Light" />
    <RadzenButton Text="Refresh" Click="(() => { ds.Refresh(); })" ButtonStyle="ButtonStyle.Light" />
    <RadzenDatePicker TValue="DateTime" />
    </div>);
```

**Expected behavior**
Prevent focus to move pass RadzenDatePicker.

**Aditional**
RadzenDropDown and RadzenDropDownDataGrid have the same problem.

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser Chrome
 - Version 114.0.5735.133
